### PR TITLE
[mini] Intrinsify Marshal.PtrToStructure<T>

### PR DESCRIPTION
**Pull request by kumpera**

### CODE REVIEW

File: mono/mini/method-to-ir.c [hunk 1]
```diff
@@ -5114,6 +5114,31 @@ mini_emit_inst_for_sharable_method (MonoCompile *cfg, MonoMethod *cmethod, MonoM
...
+	if (MONO_TYPE_IS_REFERENCE (t))
+		return FALSE;
+
+	if (MONO_TYPE_IS_PRIMITIVE_SCALAR (t))
+		return TRUE;
+
+	MonoClass *klass = mono_class_from_mono_type (t);
+
+	//MonoClass::blitable depends on mono_class_setup_fields beind done.
+	mono_class_setup_fields (klass);
```
Comment (vargaz) [inline mono/mini/method-to-ir.c]: Typo.

**Comment by vargaz** [comment]
Looks ok otherwise.

File: mono/mini/method-to-ir.c (hunk 1 above)
Comment (kumpera) [inline mono/mini/method-to-ir.c]: fixed

**Comment by kumpera** [comment]
build

**Comment by kumpera** [comment]
Failures look unrelated.

**Pull request merged**